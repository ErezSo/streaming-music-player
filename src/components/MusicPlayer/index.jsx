import React from "react";
import ReactAudioPlayer from "react-audio-player";

const styles = {
  player: {
    paddingRight: "2em",
    float: "right"
  }
};

const MusicPlayer = ({ song }) => {
  return (
    <div className="navbar-text navbar-right" style={styles.player}>
      <ReactAudioPlayer src={song} autoPlay controls />
    </div>
  );
};

export default MusicPlayer;
