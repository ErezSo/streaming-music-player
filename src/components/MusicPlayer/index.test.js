import React from "react";
import { shallow } from "enzyme";
import MusicPlayer from "./";

describe("MusicPlayer", () => {
  let wrapper;
  const song =
    "https://hearthis.at/tdsmix/bestami-turna-soundpark-14042018/listen/?s=rCW";

  beforeEach(() => {
    wrapper = shallow(<MusicPlayer song={song} />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
