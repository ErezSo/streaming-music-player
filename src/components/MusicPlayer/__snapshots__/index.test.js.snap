// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicPlayer renders correctly 1`] = `
"<div className=\\"navbar-text navbar-right\\" style={{...}}>
  <t src=\\"https://hearthis.at/tdsmix/bestami-turna-soundpark-14042018/listen/?s=rCW\\" autoPlay={true} controls={true} className=\\"\\" controlsList=\\"\\" crossOrigin=\\"\\" id=\\"\\" listenInterval={10000} loop={false} muted={false} onAbort={[Function: onAbort]} onCanPlay={[Function: onCanPlay]} onCanPlayThrough={[Function: onCanPlayThrough]} onEnded={[Function: onEnded]} onError={[Function: onError]} onListen={[Function: onListen]} onPause={[Function: onPause]} onPlay={[Function: onPlay]} onSeeked={[Function: onSeeked]} onVolumeChanged={[Function: onVolumeChanged]} onLoadedMetadata={[Function: onLoadedMetadata]} preload=\\"metadata\\" style={{...}} title=\\"\\" volume={1} />
</div>"
`;
